### ISSUE TITLE
bug introduced in .set 

### ISSUE THREAD

**Issue by s7dhansh** [jagi/meteor-astronomy#538]
The latest release has a bug. Wasn't there in 2.2.4
```
n1: {
  type: N1
}

N1 = Class.create({
  ...
 fields: {
    n2: {
      type: N2
    }
  }
});
```

`doc.set({n1: {n2: {...}}})` makes n2 an object of N1 class instead of N2 class.

**Comment by lukejagodzinski** [comment]
You have a bug in your example. You're setting the `n1` field which does not exist in the `N1` class. So it's not possible that such a code could work. I've tested and nested classes are casted properly. So it's probably bug in your code.

**Comment by s7dhansh** [comment]
Reprod: https://github.com/settlin/astro-bug

I did not explain the entire structure. It's like this.

```
N2 = Class.create({
    name: "N2",
    fields: {
        a: {
            type: Number,
            optional: true
        },
    }
});

N1 = Class.create({
    name: "N1",
    fields: {
        n2: {
            type: N2,
            optional: true
        },
    }
});

export const Post = Class.create({
    name: "Post",
    collection: Posts,
    secured: false,
    fields: {
        n1: {
            type: N1,
            default() { return {}; },
        },
    }
});
```

What fails is `post.set("n1.n2", {a: 1})`, that too only when `post.n1` is `undefined` initially.

**Comment by lukejagodzinski** [comment]
So it's correct behavior. You can't set value on a document that does not exist. Astronomy can't figure out how to create such a nested document.

**Comment by s7dhansh** [comment]
1. It worked till 2.2.4. I have not made any code changes. (It also works in mongo/minimongo).
2. Even if you plan to phase out this functionality (though I would beg you not to, as it removes all the flexibility that v2.3 provides with its cast and merge features), it should show proper error. What it does instead is that it sets `n1.n2` to be of N1 class, instead of N2 class.
   Please checkout the repo, run, click on the button and see the field and error.

**Comment by lukejagodzinski** [comment]
Ok I will investigate it later today, but for me it should not work at all. It shouldn't set any value and if it did in previous version then it was a bug.

**Comment by lukejagodzinski** [comment]
Sorry that it took me so long to investigate, but at first I had problems with running this reproduction because of the bug in Meteor and I've postponed it a little bit. Today I've took a look at it and in deed there is a bug that I've introduced... Sorry for that. I have to create more tests, to find such a bugs on earlier stages of development.

Previously I've been casting default values to its classes and with the new `set()` options I've changed some things and accidentally I've removed casting default values. So I've prepared a fix and I will release a new version today or tomorrow.
